platform/jobs/edxEndToEndRepoMasterWeeklyJob.groovy

```diff
@@ -0,0 +1,165 @@
+    default: 'AR-1000'
+    ],
+    [
+    name: 'courseRun',
+    description: 'Term in which course will run',
+    default: 'fall'
+    ],
+    [
+    name: 'courseDisp',
+    description: 'Display name of the course',
+    default: 'Manual Smoke Test Course 1 - Auto'
+    ]
+]
+
+/* Map to hold the k:v pairs parsed from the secret file */
+Map secretMap = [:]
+out.println('Parsing secret YAML file')
+try {
+    /* Parse k:v pairs from the secret file referenced by secretFileVariable */
+    String contents = new File("${EDX_END_TO_END_TESTS}").text
+    Yaml yaml = new Yaml()
+    secretMap = yaml.load(contents)
+    out.println('Successfully parsed secret YAML file')
+}
+catch (any) {
+    out.println('Jenkins DSL: Error parsing secret YAML file')
+    out.println('Exiting with error code 1')
+    return 1
+}
+
+/* Iterate over the job configurations */
+secretMap.each { jobConfigs ->
+
+    Map jobConfig = jobConfigs.getValue()
+
+    /* Test secret contains all necessary keys for this job */
+    /* TODO: Use/Build a more robust test framework for this */
+    assert jobConfig.containsKey('authUser')
+    assert jobConfig.containsKey('authPass')
+    assert jobConfig.containsKey('loginEmail')
+    assert jobConfig.containsKey('loginPass')
+    assert jobConfig.containsKey('courseOrg')
+    assert jobConfig.containsKey('courseNumber')
+    assert jobConfig.containsKey('courseRun')
+    assert jobConfig.containsKey('courseDisp')
+    assert jobConfig.containsKey('open')
+    assert jobConfig.containsKey('email')
+    assert jobConfig.containsKey('githubRepoUrlSegment')
+    assert jobConfig.containsKey('jobName')
+    assert jobConfig.containsKey('platformCredential')
+
+    job(jobConfig['jobName']) {
+
+        /* For non-open jobs, enable project based security */
+        if (!jobConfig['open'].toBoolean()) {
+            authorization {
+                blocksInheritance(true)
+                permissionAll('edx')
+            }
+        }
+
+        parameters {
+            stringParams.each { param ->
+                stringParam(param.name, param.default, param.description)
+            }
+        }
+
+        properties {
+              githubProjectUrl(JENKINS_PUBLIC_GITHUB_BASEURL + jobConfig['githubRepoUrlSegment'])
+        }
+
+        logRotator JENKINS_PUBLIC_LOG_ROTATOR() //Discard build after 14 days
+
+        label('jenkins-worker') //restrict to jenkins-worker
+
+        checkoutRetryCount(5)
+
+        environmentVariables {
+            env('BASIC_AUTH_USER', jobConfig['authUser'])
+            env('BASIC_AUTH_PASSWORD', jobConfig['authPass'])
+            env('USER_LOGIN_EMAIL', jobConfig['loginEmail'])
+            env('USER_LOGIN_PASSWORD', jobConfig['loginPass'])
+            env('COURSE_ORG', jobConfig['courseOrg'])
+            env('COURSE_NUMBER', jobConfig['courseNumber'])
+            env('COURSE_RUN', jobConfig['courseRun'])
+            env('COURSE_DISPLAY_NAME', jobConfig['courseDisp'])
```

Comment:
Again, since these values are not parsed out of secrets, you must access them differently